public/coda/coding_progress.js

```diff
@@ -19,7 +19,37 @@ function update_progress_ui(data) {
         rw.insertCell().innerText = dataset_id
         rw.insertCell().innerText = messages_count
         rw.insertCell().innerText = messages_with_label
-        rw.insertCell().innerText = (100 * messages_with_label / messages_count).toFixed(2) + "%"  
+        rw.insertCell().innerText = (100 * messages_with_label / messages_count).toFixed(2) + '%'
         console.log(dataset_id, messages_count, messages_with_label);
-      }
+    //Formating the rows based on cell value
+        $(document).ready(function(){
+            //Grab the cells of the last rows
+            $('#codingtable td:nth-child(4)').each(function() {
+               var Done = $(this).text();
+                //Style the entire row conditionally based on the cell value
+                if ((parseFloat(Done) >= 0) && (parseFloat(Done) <= 25)) {
+                    $(this).parent().addClass('coding-below25'); 
+                }
+                else if((parseFloat(Done) > 25) && (parseFloat(Done) <= 50)) {
+                    $(this).parent().addClass('coding-above25'); 
+                }
+                else if((parseFloat(Done)> 50) && (parseFloat(Done) <= 75)) {
+                    $(this).parent().addClass('coding-above50'); 
+                }
+                else if((parseFloat(Done) > 75) && (parseFloat(Done) <100)) {
+                    $(this).parent().addClass('coding-above75'); 
+                }
+                else {
+                    $(this).parent().addClass('coding-complete'); 
+                }
+            });
+        });
+        //Table sorting using tablesorter plugin based on fraction of message labelling complete
+        $(function() {
+            $("#codingtable").tablesorter({
+                //sorting on page load, column four in descending order i.e from least coded to most coded.
```

Comment:
Is sorting on page load enough? What happens if an update from Firebase comes in that changes the order? We can expect this to be the case if the page is opened for a long time

Reply:
Have added table sorting on data update.